src/OAuthTokenClient.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Razorpay.Api
+{
+    public class OAuthTokenClient : Entity
+    {
+        private readonly string clientId = "client_id";
+        private readonly string clientSecret = "client_secret";
+        private readonly string code = "code";
+        private readonly string grantType = "grant_type";
+        private readonly string refreshToken = "refresh_token";
+        private readonly string token = "token";
+        private readonly string tokenTypeHint = "token_type_hint";
+        private readonly string redirectUri = "redirect_uri";
+        private readonly string scopes = "scopes";
+        private readonly string state = "state";
+        private readonly string mode = "mode";
+
+        private readonly PayloadValidator payloadValidator;
+
+        public OAuthTokenClient()
+        {
+            payloadValidator = new PayloadValidator();
+        }
+
+        public string GetAuthUrl(Dictionary<string, object> data)
+        {
+            ValidateAuthUrlRequest(data);
```

Comment:
ValidateAuthUrlRequest etc functions seems to be overhead. inside those functions there is nothing but 1 line of code to validate. we can replace that here itself

Reply:
Updated